src/main/java/eggventory/ui/InputPredictor.java

```diff
@@ -0,0 +1,243 @@
+package eggventory.ui;
+
+import eggventory.commands.CommandDictionary;
+import java.util.ArrayList;
+
+//@@author Raghav-B
+public class InputPredictor {
+
+    private CommandDictionary commandDictionary; // Contains all valid commands+arguments that can be input.
```

Comment:
it could also just be a static class though right

Reply:
Hmm maybe, but I wasn't sure how to use the `.add()` methods of ArrayList when making it a static class since there is no constructor. Unless we have a separate method called `initialize()` in the static class that we call first instead of creating an object that would by default call the constructor.